backend/dataall/modules/worksheets/db/worksheet_repositories.py

```diff
@@ -53,3 +53,17 @@ def paginated_user_worksheets(session, username, groups, uri, data=None, check_p
             page=data.get('page', WorksheetRepository._DEFAULT_PAGE),
             page_size=data.get('pageSize', WorksheetRepository._DEFAULT_PAGE_SIZE),
         ).to_dict()
+
+    @staticmethod
+    def find_query_result_by_format(
+        session, worksheet_uri: str, athena_query_id: str, file_format: str
+    ) -> WorksheetQueryResult:
+        return (
+            session.query(WorksheetQueryResult)
+            .filter(
+                WorksheetQueryResult.worksheetUri == worksheet_uri,
+                WorksheetQueryResult.AthenaQueryId == athena_query_id,
+                WorksheetQueryResult.fileFormat == file_format,
+            )
+            .first()
+        )
```

Comment:
This query can be optimized by using the primary key of the table. We could directly use a `find_query_result_by_uri` function as 

```
    @staticmethod
    def find_worksheet_result_by_uri(session, uri: str) -> WorksheetQueryResult
        return session.query(WorksheetQueryResult).get(uri)
```

Reply:
At the point where we are requesting to download the data we do not have `uri` for `WorksheetQueryResult`,
so we have to check if the url is already generated for `worksheetUri`, `athenaQueryId` and `fileFormat`.

so the query cannot be optimized only using `uri`. 
To optimize the query we can create index on `worksheetUri` and `fileFormat`

Reply:
My bad, I understood it wrong! it is the WorksheetUri not the query results uri! Resolving